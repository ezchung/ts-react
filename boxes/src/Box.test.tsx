import React from "react";
import { render, getByText, fireEvent } from "@testing-library/react";
import Box from './Box';

const mockRemove = jest.fn();


describe('Box', function () {

    const testBox = <Box id='1' height={1} width={1} backgroundColor='blue' remove={mockRemove} />;


    it('renders without crashing', function () {
        render(testBox);
    });

    it('matches snapshot', function () {
        const { container } = render(testBox);
        expect(container).toMatchSnapshot();
    });

    it('calls remove on button click', function () {
        const { container } = render(testBox);
        const button = container.querySelector('.Box-removeBtn') as HTMLElement;
        fireEvent.click(button);
        expect(mockRemove).toHaveBeenCalledTimes(1);
    });


});