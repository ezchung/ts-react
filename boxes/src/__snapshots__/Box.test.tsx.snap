// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 1em; width: 1em; background-color: blue;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
